### ISSUE TITLE
Unknown compiler option 'jsxFragmentFactory'

### ISSUE THREAD

**Issue by TechQuery** [developit/microbundle#711]
I guess MicroBundle hasn't support TypeScript 4.x, because I got an error:
> rpt2: config error TS5023: Unknown compiler option 'jsxFragmentFactory'.

**Comment by wardpeet** [comment]
Thanks will look into it 👍 

Could you create a small comment on how to reproduce this? Or do you have a repo?

**Comment by TechQuery** [comment]
Just add an option `"jsxFragmentFactory": "Fragment"` in your `tsconfig.json`, and then run `microbundle` again, you'll see the error.

**Comment by developit** [comment]
We should check TS version and use the appropriate config - it would be a shame to have Microbundle only work with TS 4.

**Comment by whitetrefoil** [comment]
Not only tsconfig.json options but also code syntax will have the same ts version problem.

In my case:
```ts
try {
  something()
} catch (e: unknown) {
         ^ (rpt2 plugin) Error: semantic error TS1196: Catch clause variable cannot have a type annotation.
  something()
}
```

Where here `(e: unknown)` is [new syntax in TS 4.0](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-0.html#unknown-on-catch-clause-bindings), it's recommanded (safer) and enforced by eslint.

Maybe microbundle can try locally installed ts first before using rpt2's default one.  Or at least some way to set the version manually.

**Comment by developit** [comment]
Thanks for the idea and PR @whitetrefoil - good thinking.

**Comment by developit** [comment]
Fixed in #738.
